Ticketing/Ticketing.UI/Controlers/EventsController.cs

```diff
@@ -0,0 +1,47 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using Ticketing.BAL.Contracts;
+using Ticketing.BAL.Model;
+using Ticketing.BAL.Services;
+using Ticketing.DAL.Domain;
+
+namespace Ticketing.UI.Controlers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class EventsController : ControllerBase
+    {
+        private IEventService _eventService;
+
+        public EventsController(IEventService eventService)
+        {
+            _eventService = eventService;
+        }
+
+        //GET /events
+        [HttpGet]
+        public async IAsyncEnumerable<EventReturnModel> Get()
+        {
+            var events = await _eventService.GetEventsAsync();
+
+            foreach (var @event in events)
+            {
+                yield return @event;
```

Comment:
do you really need this yeld, and how you will return like example 204 status code?

Reply:
done